contrib/inventory/azure_rm.py

```diff
@@ -548,23 +550,35 @@ def compute_client(self):
             self._register('Microsoft.Compute')
         return self._compute_client
 
+    @property
+    def subscription_client(self):
+        self.log('Getting subscription')
+        if not self._subscription_client:
+            self._subscription_client = SubscriptionClient(
```

Comment:
call `get_mgmt_client()` to create a client?

Reply:
This method did not exist when I made this patch. I'll update the PR.

Reply:
How about change the `get_mgmt_svc_client()` method as 

```python
    def get_mgmt_svc_client(self, client_type, base_url, api_version):
        client_kwargs = dict(credentials=self.azure_credentials, subscription_id=self.subscription_id, base_url=base_url)
        if api_version:
            client_kwargs['api_version'] = api_version
        client = client_type(**client_kwargs)
        client.config.add_user_agent(ANSIBLE_USER_AGENT)
        return client
```

Since we want to construct the client instance in a single entry.